### DISCUSSION TITLE
Where can I get sn-pulsar-tools?

### DISCUSSION THREAD

**Discussion by Raven888888** [apache/pulsar#18083] [Q&A]
I am using the open sourced pulsar binary, but recently bookkeeper ledgers has grown to a large size.

Come across this pulsar consistency check ([PCK](https://docs.streamnative.io/platform/latest/operator-guides/configure/sn-pulsar-tool/pck-overview)) tool that is supposed to clean orphan and inactive topic ledgers.

Where can I get the tool to use it on my pulsar cluster?

**Comment by danielnesaraj** [comment]
We faced this issue once as well. Having an admin client command or a separate CLI tool would be very helpful here! We ended up blindly rolling restart all consumers, bookies, and brokers to resolve the issue and we still have no idea what the root cause is.
